File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.cpp
class Solution {
public:
    int numberOfSpecialChars(string word) {
        bool capitals[26];
        bool smalls[26];
        fill_n(capitals, 26, false);
        fill_n(smalls, 26, false);

        for(const char &ch : word){
            if(ch - 'a' < 0) capitals[ch-'A'] = true;
            else smalls[ch-'a'] = true;
        }

        int count = 0;
        for(int i = 0; i < 26; i++){
            if(smalls[i] and capitals[i]) count++;
        }

        return count;
    }
};

File: 3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.java
class Solution {
    public int numberOfSpecialChars(String word) {
        boolean[] lowerCase = new boolean[26];
        boolean[] upperCase = new boolean[26];

        for(char ch : word.toCharArray()){
            if(ch >= 'a' && ch <= 'z') lowerCase[(int)(ch - 'a')] = true;
            else upperCase[(int)(ch - 'A')] = true;
        }

        int count = 0;
        for(int i = 0; i < 26; i++){
            if(lowerCase[i] && upperCase[i]) count++;
        }

        return count;
    }
}